### ISSUE TITLE
DialogFragment's show is not analyzed correctly in test02

### ISSUE THREAD

**Issue by LuxiWang99** [izgzhen/markii#54]
The Diapromt button in test02 prompts a dialog but this dialog is not included in dialogView.facts, nor are the buttons contained in this dialogViewButton.facts. 

Likely because the dialog is created differently from the one in test01.

**Comment by LuxiWang99** [comment]
The reason why the dialog is not analyzed is that the dialog is created within a class (a subclass of DialogFragment). The dialog builder is created within this class, but this class is not analyzed.

**Comment by izgzhen** [comment]
As a first step, let's check in the test02 with current partial facts first. (open a PR for this)

**Comment by izgzhen** [comment]
You will probably need to work on `com.research.nomad.markii.analyses.PreVASCO#analyze`.

First, you will probably need to mimick this part for `DialogFragment`:

```java
             if (Constants.isDialogShow(invokedMethod.getSignature) ||
                Constants.isDialogBuilderShow(invokedMethod.getSignature)) {
                val dialogBase = invokeExpr.asInstanceOf[InstanceInvokeExpr].getBase.asInstanceOf[Local]
                val methods = mutable.Set[SootMethod]()
                for (defStmt <- PreAnalyses.getDefsOfAt(reached, dialogBase, stmt)) {
                  DynamicCFG.getRunnerOfDialog(defStmt) match {
                    case Some(runner) => methods.add(runner.method)
                    case None =>
                  }
                }
                if (methods.nonEmpty) {
                  // FIXME: I need to build a static class for this type of work
                  val runAllMethod = DynamicCFG.getRunAllDialog(stmt, methods, reached)
                  val invocation = Jimple.v().newInvokeStmt(Jimple.v().newStaticInvokeExpr(runAllMethod.makeRef(), dialogBase))
                  Scene.v().getCallGraph.removeAllEdgesOutOf(stmt)
                  Scene.v().getCallGraph.addEdge(new Edge(reached, invocation, runAllMethod))
                  swaps.put(stmt, invocation)
                  startWindowStmts.add(invocation)
                  showDialogInvocations.put(invocation, dialogBase)
                }
              }
```

Note that you can use `AppInfo.hier.isSubclassOf` for the outermost `if` check.

Inside the part, you will probably need to write a new thing (e.g. called `getRunnerOfDialogFragment`) similar to `getRunnerOfDialog` to make it put the corresponding `onCreateDialog` inside the created fake method. The new `getRunnerOfDialogFragment`will probably contains some code similar to ones in `getRunner` as well.

**Comment by izgzhen** [comment]
Discussion summary:

* First, the inline click handlers (XML-declared ones) are probably not processed *during* VASCO (though it is process post-VASCO). We need to process them during VASCO so it appears in the call-graph for further analysis
* Second, the further analysis of the "show" API of `DialogFragment` need to happen during VASCO as well (inside `callLocalFlowFunction`), for it to sequentially update the call-graph with the `DialogFragment` instance's lifecycle method invocations.

**Comment by izgzhen** [comment]
I will work on the first point. It might mean that we can delete the post-VASCO processing of inline click handlers, but I might not investigate whether this is true by myself.

**Comment by izgzhen** [comment]
Fixed by #60
